### PR TITLE
Add additional plotting tests

### PR DESCRIPTION
**Pull request by akaszynski**
Add a handful of plotting tests and improves or splits several of the volume plotting image regression tests.

### CODE REVIEW

File: tests/plotting/test_plotting.py [hunk 1]
```diff
@@ -1646,32 +1646,43 @@ def test_image_properties():
...
     data['d'].rename_array('Spatial Point Data', 'd')
-    data.plot(volume=True, multi_colors=True)
+    data.plot(volume=True, multi_colors=True, before_close_callback=verify_cache_image)
+
 
-    # Check that NumPy arrays work
-    arr = vol["Spatial Point Data"].reshape(vol.dimensions)
+def test_array_volume_rendering(uniform):
+    arr = uniform["Spatial Point Data"].reshape(uniform.dimensions)
     pyvista.plot(arr, volume=True, opacity='linear')
```
[the remark this answers is not in the document]
Comment (akaszynski) [reply tests/plotting/test_plotting.py:1686]: Yes, thanks for that.

File: pyvista/plotting/charts.py [hunk 2]
```diff
@@ -1364,6 +1364,7 @@ def show(
         window_size=None,
         notebook=None,
         background='w',
+        **kwargs,
```
Comment (banesullivan) [inline pyvista/plotting/charts.py:1367]: Oof, I am really not a fan of unchecked `**kwargs`, these can really come back to bite us. If they are truly for developer use, then could we make this `developer_kwargs=None` instead?

File: tests/plotting/test_plotting.py [hunk 3]
```diff
@@ -1646,33 +1646,44 @@ def test_image_properties():
     p.close()
 
 
-def test_volume_rendering():
-    # Really just making sure no errors are thrown
-    vol = examples.load_uniform()
-    vol.plot(volume=True, opacity='linear')
+def test_volume_rendering_from_helper(uniform):
+    uniform.plot(volume=True, opacity='linear', before_close_callback=verify_cache_image)
```
Comment (banesullivan) [inline tests/plotting/test_plotting.py:1650]: We should really just make a pytest fixture context that sets `verify_cache_image` as the default `before_close_callback`... to do in a later PR though

**Review by banesullivan — CHANGES_REQUESTED** [review]

File: pyvista/plotting/charts.py (hunk 2 above)
Comment (akaszynski) [reply pyvista/plotting/charts.py]: Resolved in 41ae933.

**Review by banesullivan — DISMISSED** [review]

**Pull request merged**